samples/client/petstore/java/apache-httpclient/api/openapi.yaml

```diff
@@ -1507,15 +1507,18 @@ components:
       - $ref: '#/components/schemas/Cat_allOf'
     Animal:
       discriminator:
-        propertyName: className
+        mapping:
+          DOG: '#/components/schemas/Dog'
+          CAT: '#/components/schemas/Cat'
+        propertyName: type
       properties:
-        className:
+        type:
```

Comment:
Same nit - would probably call this `animalType` or `animalSpecies`

Reply:
changed to `species`

Reply:
i've cherry-pick some commits and created https://github.com/OpenAPITools/openapi-generator/pull/14984 instead.

i didn't rename `className` to something else